packages/flame_forge2d/example/lib/balls.dart

```diff
@@ -60,39 +60,63 @@ class Ball extends BodyComponent {
       }
     }
   }
+
+  @override
+  void beginContact(Contact contact) {
+    final otherObject = contact.bodyA.userData == this
+        ? contact.bodyB.userData
+        : contact.bodyA.userData;
+
+    if (otherObject is Wall) {
+      otherObject.paint = paint;
+    }
+
+    if (otherObject is WhiteBall) {
+      return;
+    }
+
+    if (otherObject is Ball) {
+      if (paint != originalPaint) {
+        paint = otherObject.paint;
+      } else {
+        otherObject.paint = paint;
+      }
+    }
+  }
+
+  @override
+  void endContact(Contact contact) {}
+
+  @override
+  void postSolve(Contact contact, ContactImpulse impulse) {}
+
+  @override
+  void preSolve(Contact contact, Manifold oldManifold) {}
```

Comment:
This can be removed once https://github.com/flame-engine/forge2d/pull/55 is merged.